source/DistanceAndDirection/ArcMapAddinDistanceAndDirection/ArcMapAddinDistanceAndDirection/ViewModels/TabBaseViewModel.cs

```diff
@@ -1071,6 +1071,51 @@ internal void AddTextToMap(IGeometry geom, string text)
             RaisePropertyChanged(() => HasMapGraphics);
         }
 
+        internal void AddTextToMap(IGeometry geom, string text, double angle, AzimuthTypes azimuthType)
+        {
+            var mxDoc = ArcMap.Application.Document as IMxDocument;
+            var av = mxDoc.FocusMap as IActiveView;
+            var gc = av as IGraphicsContainer;
+            double bearing = 0.0;
+            if(azimuthType == AzimuthTypes.Mils)
```

Comment:
I think all of the existing methods also convert from radians so I just put the formulas in this piece of code.